Source/CombatExtended/CombatExtended/Comps/CompMeleeTargettingGizmo.cs

```diff
@@ -0,0 +1,254 @@
+
+                //just in case of attacking some weird creature
+                if (torso != null)
+                {
+                    var torsoApparel = target.apparel.WornApparel.FindAll(x => x.def.apparel.CoversBodyPart(torso));
+
+                    float overallRHA = 0f;
+
+                    if (torsoApparel.Count() > 0)
+                    {
+                        foreach (Apparel apparel in torsoApparel)
+                        {
+                            overallRHA += apparel.GetStatValue(StatDefOf.ArmorRating_Sharp);
+                        }
+                    }
+                    if (maxWeaponPen < overallRHA)
+                    {
+                        return BodyPartHeight.Top;
+                    }
+                }
+
+                return BodyPartHeight.Middle;
+            }
+
+            if (PawnParent.skills.GetSkill(SkillDefOf.Melee).Level > 16)
+            {
+                targetBodyPart = BodyPartDefOf.Neck;
+
+                var neck = target.health.hediffSet.GetNotMissingParts(BodyPartHeight.Top).Where(y => y.def == BodyPartDefOf.Neck).FirstOrFallback();
+
+                if (neck != null)
+                {
+                    var neckApparel = target.apparel.WornApparel.Find(x => x.def.apparel.CoversBodyPart(neck));
+
+                    if (neckApparel == null | maxWeaponPen < neckApparel.GetStatValue(StatDefOf.ArmorRating_Sharp))
+                    {
+                        Log.Message(neckApparel.Label);
+                        targetBodyPart = null;
+                    }
+                    else
+                    {
+                        targetBodyPart = neck.def;
+                    }
+                }
+
+                return BodyPartHeight.Top;
+            }
+
+            return BodyPartHeight.Undefined;
+        }
+            
+
+
+        public bool SkillReqA
+        {
+            get
+            {
+                return PawnParent.skills.GetSkill(SkillDefOf.Melee).Level >= 5;
+            }
+        }
+        public bool SkillReqBP
+        {
+            get
+            {
+                return PawnParent.skills.GetSkill(SkillDefOf.Melee).Level > 15;
+            }
+        }
+
+        public float SkillBodyPartAttackChance
+        {
+            get
+            {
+                var result = 0.20f;
+
+                result *= ((PawnParent.skills.GetSkill(SkillDefOf.Melee).Level * 1f) - 15f);
+
+                return result;
+            }
+        }
+
+        private string heightString
+        {
+            get
+            {
+                var result = heightInt.ToString();
+
+                if (result.ToLower() == "undefined")
+                {
+                    result = "Automatic";
+                }
+
+                return result;
+            }
+        }
+
+        #endregion
+
+
+        #region Methods
+        public override IEnumerable<Gizmo> CompGetGizmosExtra()

```

Comment:
Same as the other PR, it would be better to used localized strings for all of the UI parts.